src/ValidatedObjectTrait.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Input\Validation;
+
+use Yiisoft\Hydrator\Attribute\SkipHydration;
+use Yiisoft\Validator\Result;
+
+trait ValidatedObjectTrait
+{
+    #[SkipHydration]
+    private ?Result $validationResult = null;
+
+    public function processValidationResult(Result $result): void
```

Comment:
```suggestion
    public function setValidationResult(Result $result): void
```

Reply:
It's implementation of `PostValidationHookInterface` from Yii Validator.